catalog/ui/catalog-ui-search/src/main/webapp/component/location-old/location-old.view.js

```diff
@@ -426,10 +433,77 @@ define([
             this.$('#utmZone').multiselect(singleselectOptions);
             this.$('#utmHemisphere').multiselect(singleselectOptions);
 
+            var keywordProperty = new Property({
+                placeholder: 'Enter a region, country, or city',
+                minimumInputLength: 2,
+                url: '/search/catalog/internal/geofeature/suggestions',
+                type: 'AUTOCOMPLETE'
+            });
+            this.listenTo(keywordProperty, 'change:value', this.handleGetKeyword);            
+            var PropertyView = require('component/property/property.view');
+            var keywordPropertyView = new PropertyView({ model: keywordProperty });            
+            keywordPropertyView.turnOnLimitedWidth();
+            this.keyword.show(keywordPropertyView);
+
             this.blockMultiselectEvents();
             this.updateLocationFields();
             this.handleEdit();
         },
+        handleGetKeyword: function(model, values) {
+            var query = values[0];
+            if (!query) return;
+
+            var view = this;
+            if (view.getFeatureByKeywordXHR) {
+                view.getFeatureByKeywordXHR.abort();
+            }
+            view.getFeatureByKeywordXHR = $.get({
+                url: '/search/catalog/internal/geofeature?name=' + query,
+                contentType: 'application/json',
+                cache: false
+            }).done(function(data){
+                view.showKeywordResults(data);
+            }).always(function(){
+                view.getFeatureByKeywordXHR = null;
+            });
+        },
+        showKeywordResults: function(data) {
+            var eventToDrawShape = null;
+            var attrsToSet = null;
+
+            switch(data.type) {
+                case "bbox": {
+                    var { north, south, west, east } = data;
+                    attrsToSet = { north, south, west, east };
+                    eventToDrawShape = 'search:bboxdisplay';                    
+                    break;
+                }
+                case "polygon": {
+                    var polygon = data.coordinates.map(function(coord){
+                        return [coord.latitude, coord.longitude];
+                    });
+                    attrsToSet = { polygon, bbox: undefined };
+                    eventToDrawShape = 'search:polydisplay';
+                    break;
+                }
+                case "point-radius": {
+                    var lat = data.center.latitude;
+                    var lon = data.center.longitude;
+                    var radius = data.radius;
+                    attrsToSet = { lat, lon, radius, bbox: undefined };
```

Comment:
There's more code running than what is seen in this block, though, because events get triggered and handlers run (which then auto-magically fill in fields in the DOM). 
I mispoke/forgot the actual symptom - searching for a bounding box feature first and then a different feature type afterwards (could happen once I add country polygons) results in an invalid polygon being saved (all NaNs). I believe it has to do with existing code that converts any bounding box (include one you draw yourself) into a polygon on save.